shell/platform/windows/flutter_windows.cc

```diff
@@ -0,0 +1,242 @@
+                                       const char* what) -> void* {
+    return eglGetProcAddress(what);
+  };
+
+  FlutterProjectArgs args = {};
+  args.struct_size = sizeof(FlutterProjectArgs);
+  args.assets_path = assets_path;
+  args.icu_data_path = icu_data_path;
+  args.command_line_argc = static_cast<int>(argv.size());
+  args.command_line_argv = &argv[0];
+  args.platform_message_callback =
+      [](const FlutterPlatformMessage* engine_message,
+         void* user_data) -> void {
+    auto window = reinterpret_cast<flutter::Win32FlutterWindow*>(user_data);
+    return window->HandlePlatformMessage(engine_message);
+  };
+
+  // args.custom_task_runners = custom_task_runners; TODO
+
+  FLUTTER_API_SYMBOL(FlutterEngine) engine = nullptr;
+  auto result =
+      FlutterEngineRun(FLUTTER_ENGINE_VERSION, &config, &args, window, &engine);
+  if (result != kSuccess || engine == nullptr) {
+    std::cerr << "Failed to start Flutter engine: error " << result
+              << std::endl;
+    return nullptr;
+  }
+
+  return engine;
+}
+
+bool FlutterDesktopInit() {
+  return true;
+}
+
+void FlutterDesktopTerminate() {}
+
+FlutterDesktopWindowControllerRef FlutterDesktopCreateWindow(
+    int initial_width,
+    int initial_height,
+    const char* title,
+    const char* assets_path,
+    const char* icu_data_path,
+    const char** arguments,
+    size_t argument_count) {
+  auto win = std::make_unique<flutter::Win32FlutterWindow>(
+      title, 10, 10, initial_width, initial_height);
+
+  flutter::Win32FlutterWindow* window = win.get();
+
+  auto engine = RunFlutterEngine(window, assets_path, icu_data_path, arguments,
+                                 argument_count);
+
+  if (engine == nullptr) {
+    return nullptr;
+  }
+
+  auto state = window->SetState(engine);
+  state->window = std::move(win);
+
+  // Trigger an initial size callback to send size information to Flutter.
+  window->SendWindowMetrics();
+
+  return state;
+}
+
+void FlutterDesktopDestroyWindow(FlutterDesktopWindowControllerRef controller) {
+  FlutterEngineShutdown(controller->engine);
+  delete controller;
+}
+
+void FlutterDesktopWindowSetHoverEnabled(FlutterDesktopWindowRef flutter_window,
+                                         bool enabled) {
+  // todo either implement or remove once embedder project has moved
+}
+
+void FlutterDesktopWindowSetTitle(FlutterDesktopWindowRef flutter_window,
+                                  const char* title) {
+  // todo either implement or remove
+}
+
+void FlutterDesktopWindowSetIcon(FlutterDesktopWindowRef flutter_window,
+                                 uint8_t* pixel_data,
+                                 int width,
+                                 int height) {
+  // todo either implement or remove
+}
+
+void FlutterDesktopRunWindowLoop(FlutterDesktopWindowControllerRef controller) {
+  controller->window->FlutterMessageLoop();
+
+  FlutterEngineShutdown(controller->engine);
```

Comment:
Call `FlutterDesktopDestroyWindow` rather than duplicate code?

Reply:
Fixed in next PR